engine/flink/components/dev-table/src/main/scala/pl/touk/nussknacker/engine/flink/table/source/TableSource.scala

```diff
@@ -86,9 +99,105 @@ class TableSource(
 //  TODO: add implementation during task with test from file
   override def testRecordParser: TestRecordParser[RECORD] = ???
 
+  override def generateTestData(size: Int): TestData = {
+    val classLoader = getClass.getClassLoader
+
+    // setting context classloader because Flink in multiple places relies on it and without this temporary override it doesnt have
+    // the necessary classes
+    ThreadUtils.withThisAsContextClassLoader(classLoader) {
+      // TODO: check if this miniCluster env releases memory properly and if not, refactor to reuse one minicluster per all usages
+      val tableEnv = TableEnvironment.create(
+        EnvironmentSettings.newInstance.withConfiguration(miniClusterTestingEnvConfig).build()
+      )
+
+      tableEnv.createTable(
+        dataGenerationInputInternalTableName,
+        TableDescriptor
+          .forConnector("datagen")
+          .option("number-of-rows", size.toString)
+          .schema(tableDefinition.toFlinkSchema)
+          .build()
+      )
+
+      val tempDir    = Files.createTempDirectory("tableSourceDataDump-")
+      val tempDirUrl = tempDir.toUri.toURL
+      logger.debug(s"Created temporary directory for dumping test data at: '$tempDirUrl'")
+
```

Comment:
I've extracted some methods, but its not going to be scala idiomatic since its mostly interacting with Flink's API where it just returns Unit. 

I think this class will change a lot after we add support for testing on data from the generator to accomodate reuse, but thats for another PR.